### PR TITLE
Support $ref alongside description in OpenAPI V3

### PR DESCRIPTION
**Pull request by Jefftree**
Copied from the latest OpenAPI 3.1 spec: https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/schema.json#L1072-L1081

Add support for $ref to have a description alongside it in the same object in the proto model for OpenAPI V3.0 and V3.1. This is an expected property for 3.1 and a relaxed constraint for 3.0.

/assign @timburks

### CODE REVIEW

File: openapiv3/OpenAPIv3.proto [hunk 1]
```diff
@@ -208,8 +208,8 @@ message Info {
   Contact contact = 4;
   License license = 5;
   string version = 6;
-  repeated NamedAny specification_extension = 7;
-  string summary = 8;
+  string summary = 7;
```
Comment (Jefftree) [inline openapiv3/OpenAPIv3.proto:211]: This change was generated automatically after running the regeneration script. Is this expected?

Comment (timburks) [reply openapiv3/OpenAPIv3.proto:211]: I've been manually changing this back to maintain backwards-compatibility of the protos. Could you do that here? Essentially I'm asking for the summary tag to be 7 and specification_extension should be 8.

Comment (timburks) [reply openapiv3/OpenAPIv3.proto:211]: That would also require regenerating OpenAPIv3.pb.go

Comment (Jefftree) [reply openapiv3/OpenAPIv3.proto]: Fixed. Updated both the .proto and .pb.go file.

**Review by timburks — APPROVED** [review]
@Jefftree Thanks for these changes and updates! LGTM, will merge

**Pull request merged**